pkg/reconciler/workload/resource/resource_controller.go

```diff
@@ -448,7 +448,7 @@ func locations(annotations, labels map[string]string, skipPending bool) (locatio
 	}
 	for k := range annotations {
 		if strings.HasPrefix(k, workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix) {
-			deleting.Insert(strings.TrimPrefix(k, workloadv1alpha1.ClusterResourceStateLabelPrefix))
+			deleting.Insert(strings.TrimPrefix(k, workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix))
```

Comment:
Do we have a unit test for this fix?

Reply:
Yes, indirectly. The additional test added in commit 0bcf6f345ecf70f64ad1c28148c2d61b226cdcfd will fail without this fix.
So I didn't think useful to add a new dedicated test for the `locations` function